### PR TITLE
fix(slint)!: Change slint tree-sitter to official slint repo

### PR DESCRIPTION
**Pull request by jrmoulton**
Points slint to the official repository as I haven't maintained the other one in a while. 

Marking as a draft for now because I believe that slint needs to add the generated c source files first. 

@hunger I've marked you as the maintainer. Does that work?

### CODE REVIEW

**Comment by clason** [comment]
@theHamsta

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1696,10 +1696,10 @@ list.slang = {
 
 list.slint = {
   install_info = {
-    url = "https://github.com/jrmoulton/tree-sitter-slint",
-    files = { "src/parser.c" },
+    url = "https://github.com/slint-ui/slint",
+    files = { "editors/tree-sitter-slint/src/parser.c" },
```
Comment (clason) [inline lua/nvim-treesitter/parsers.lua:1700]: you should set the `location` field instead.

**Comment by clason** [comment]
> Marking as a draft for now because I believe that slint needs to add the generated c source files first.

You can always set `generate_from_grammar = true` (check other parsers in this list).

**Comment by clason** [comment]
Also, if the queries remain unchanged, this is not a breaking change.

**Comment by clason** [comment]
can you update the lockfile and README entries?

And do you expect the parser to be committed to the repo in the near future?

**Comment by clason** [comment]
@hunger looks like there _were_ breaking changes in the grammar, so the (highlight and indent) queries need to be adapted.

**Comment by clason** [comment]
Actually, the grammar looks ver different; you'll need to completely rewrite the queries.

**Comment by clason** [comment]
>  I assumed nvim-treesitter would have some config somewhere to get them from there.

That is very wrong. Queries are editor-specific, so you need to adapt the ones _in this repo_.

**Comment by clason** [comment]
I've already moved the SHA; that's how the tests are failing and I found out.

**Comment by clason** [comment]
>  Would it be OK to add an injection into the rust queries?

Sure.

**Comment by lucario387** [comment]
For only `nvim-treesitter`, `locals.scm` doesn't do anything. It's used in nvim-treesitter-refactor and possibly other plugins, so it is kept.

**Comment by clason** [comment]
the official support channel is https://matrix.to/#/#nvim-treesitter:matrix.org

**Comment by clason** [comment]
Best rebase on master yet again; I changed the macos workflow to use M1 so the required checks changed.

**Comment by clason** [comment]
If it's easier for you, you can just open a new PR from your branch; I don't mind.

**Pull request closed without merging**